### PR TITLE
CONN-645:Fix SAML Issuer when Format is Entity and SPProvidedId is given

### PR DESCRIPTION
**Pull request by alameluchidambaram**

### CODE REVIEW

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java [hunk 1]
```diff
@@ -159,6 +159,16 @@
...
             }
+            
+            if(assertion.getSamlIssuer()!= null) {
+                if(NullChecker.isNotNullish(assertion.getSamlIssuer().getIssuer())) {
+                    requestContext.put(NhincConstants.ASSERTION_ISSUER_PROP, assertion.getSamlIssuer().getIssuer());
+                }
+                if(NullChecker.isNotNullish(assertion.getSamlIssuer().getIssuerFormat())) {
+                     requestContext.put(NhincConstants.ASSERTION_ISSUER_FORMAT_PROP, assertion.getSamlIssuer()
+                        .getIssuerFormat());
+            }
```
Comment (msweaver) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java]: this is not properly formatted.

Comment (alameluchidambaram) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/saml/extraction/SamlTokenCreator.java]: Thx Matt! I will fix it.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CONNECTSamlAssertionValidator.java [hunk 2]
```diff
@@ -118,13 +120,36 @@ protected void validateAssertion(AssertionWrapper assertion) throws WSSecurityEx
...
+                }
+
+                if (!StringUtils.isBlank(issuer.getSPNameQualifier())) {
+                    attributeNotRequired = issuer.getSPNameQualifier();
+
+                }
+            }
+            if (NullChecker.isNotNullish(attributeNotRequired)) {
+
+                throw new WSSecurityException("SOAP header element Security/Assertion/Issuer/@Format = " + ""
```
Comment (msweaver) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CONNECTSamlAssertionValidator.java]: I think this error message is a little misleading. Can we just throw the exception 3 times from each of the above if conditions? That way you can specifically say SPProviderID (or NameQualifier, or SPNameQualifier) cannot be present if issuer format is entity.

Comment (alameluchidambaram) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CONNECTSamlAssertionValidator.java]: I can throw for each one of the field specifically.

**Pull request merged**